File: 백준/Silver/6603. 로또/로또.py
def combination(n, k):
    li = []
    result = []

    def recur():
        if len(li) == k:
            result.append(li[:])
            return

        for i in range(n):
            if not li or i > li[-1]:
                li.append(i)
                recur()
                li.pop()

    recur()
    return result


def main(n, arr):
    for li in combination(n, 6):
        print(" ".join([str(arr[i]) for i in li]))


import sys

input = sys.stdin.readline
while True:
    n, *arr = [int(i) for i in input().split()]

    main(n, arr)

    if n == 0:
        break

    print()

File: 백준/Silver/6603. 로또/로또.js
const fs = require('fs');
const arr = fs.readFileSync('dev/stdin').toString().trim().split("\n");
arr.pop()

const combi = (n, k, callback = () => {}) => {
    const li = [];
    const recur = () => {
        if (li.length === k){
            return callback(li);
        }
        
        for(let i = 0; i < n; i++ ){
            if(li.length == 0 || i > li.at(-1)){
                li.push(i)
                recur()
                li.pop()
            }
        }
    }
    
    recur();
}

const solve = (s) => {
    const [_, ...array] = s.split(" ").map(Number)
    combi(array.length, 6, (li) => {
        console.log(...li.map(i => array[i]));
    });
}

const main = () => {
    for (let i = 0; i < arr.length; i++) {
        solve(arr[i]);
        if (i != arr.length - 1){
            console.log();
        }
    }    
}

main()